ramp4-pcar4/ramp4-pcar4
# Improve mapnav for smaller screen heights

### Related Item(s)
Issue #2084 

### Changes
- [FEATURE] Hide most mapnav buttons at small heights, and add a button when they're hidden to access them. 
  - Mapnav buttons in the 'drawer' will wrap as needed, ensuring they're visible down to 256px screen height.


### Notes
- Wrapping occurs at fixed heights.
- Hidden items should be fully accessible with keyboard.

Normal mapnav:
<img width="382" alt="image" src="https://github.com/user-attachments/assets/e9f7abbb-a81f-4d6c-b1df-01e5274e31cf">

Small screen:
<img width="377" alt="image" src="https://github.com/user-attachments/assets/0b7120da-b20b-4e97-841b-8cf66ef646a6">

Smaller screen wrap:
<img width="303" alt="image" src="https://github.com/user-attachments/assets/f7d59568-b5c1-46d2-ae42-c9d496947dd2">

Smallest wrap:
<img width="234" alt="image" src="https://github.com/user-attachments/assets/dcb0ebc9-3f19-45d8-992e-f7bc830d9ffd">

### QA Testing
Please test against `main` branch https://ramp4-pcar4.github.io/ramp4-pcar4/main/demos/enhanced-samples.html

### Testing

Steps:
1. Go to any sample with a mapnav (default one works).
2. Make the screen smaller until the mapnav shrinks. Click the "Open navigation" button to reveal the buttons (or use the keyboard). All buttons should still work.
3. Shrink the screen further. See how the items wrap to remain visible. All buttons should remain visible down to 256px.
